src/org/benf/cfr/reader/bytecode/analysis/parse/expression/ArithmeticMonOperation.java

```diff
@@ -58,6 +62,83 @@ public Dumper dumpInner(Dumper d) {
         return d;
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        Literal l = lhs.getComputedLiteral(display);
+        if (l == null || !(l.getValue().getValue() instanceof Number))
```

Comment:
as silly as it seems, I'd be tempted to avoid the Number path, and use the type that the Literal tells you it is.  I suspect it may be a bit more boiler plate, but it's probably less liable to lead to odd typing edge cases.

Reply:
I used this as a quick way to assert that a value was there of a supported type since `instanceof` also null checks. It means I can safely cast later too, so I'd assume I've have to make this test somewhere regardless.